src/Symfony/Component/HttpClient/Tests/LegacyCachingHttpClientTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpClient\CachingHttpClient;
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\HttpClient\Response\MockResponse;
+use Symfony\Component\HttpKernel\HttpCache\Store;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @group legacy
+ */
+class LegacyCachingHttpClientTest extends TestCase
+{
+    /**
+     * @expectedDeprecation Since symfony/http-client 7.3: Passing a "Symfony\Component\HttpKernel\HttpCache\StoreInterface" as constructor's 2nd argument of "Symfony\Component\HttpClient\CachingHttpClient" is deprecated, "Symfony\Component\Cache\Adapter\TagAwareAdapterInterface" expected.
+     */
```

Comment:
If I remember correctly, `@expect...` annotation are deprecated. Instead, you should `use ExpectDeprecationTrait` in your test class, then call `$this->expectDeprecation('Since symfony/http-client 7.3: ...')` in each test case.

Reply:
Adressed